File: Python/Exercicio21.py
# Algoritmo que lê 10 valores inteiros e positivos e:
# Encontra o maior valor, o menor valor e calcula a média dos números lidos.

# Variáveis
maior = -999
menor = 999
soma = 0

# Processamento
for i in range(1, 11):
    numero = int(input("Escreva um número: "))
    if numero > maior:
        maior = numero
    if numero < menor:
        menor = numero
    if numero > 0:
        soma = soma + numero

media = soma / 10

# Saída
print("O maior número é: {0}".format(maior))
print("O menor número é: {0}".format(menor))
print("A média dos números é: {0}".format(media))

File: C/Exercicio21.c
//Algoritmo que lê 10 valores inteiros e positivos.
//Encontra o maior valor, o menor e calcula a média dos números lidos.
#include <stdio.h>

int main() {
    //Variáveis
    int n, i, maior = -999, menor = 999, soma = 0;
    float media;

    for (i = 0; i < 10; i++) {
            //Entrada
            printf("Digite um número: \n");
            scanf("%d", &n);
        //Processamento
        if (n > 0) {
            if (n > maior) {
                maior = n;
            }
            if (n < menor) {
                menor = n;
            }
            soma = soma + n;
        }else{
            printf("Número inválido. O número deve ser inteiro e positivo.\n");
            i--; //Decrementa para repetir a iteração
        }
    }
    media = soma / 10;
    //Saida
    printf("O maior número é: %d \n", maior);
    printf("O menor número é: %d \n", menor);
    printf("A média dos números é: %2.f \n", media);
    return 0;
}